tests/expected/intrinsics/simd-invalid-conversion/expected

```diff
@@ -0,0 +1,2 @@
+Kani unexpectedly panicked during compilation
```

Comment:
Is this expected? Can we make this a user friendly error?

Reply:
The problem here is that we have to do the type-checking... Compiling this with `rustc` would cause an error, but doing so with Kani doesn't. I've added cases for this in `vector_cmp` and a couple of test cases. But in there I don't have anything to exit gracefully as we do in other cases (e.g., the session object). I agree with you this should just be a compiler error, but in general this is what we do when type-checking (assertions).